src/main/java/appeng/client/me/FluidRepo.java

```diff
@@ -0,0 +1,253 @@
+import appeng.util.prioritylist.IPartitionList;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 22/05/2018
+ * @since rv6 22/05/2018
+ */
+public class FluidRepo
+{
+	private final IItemList<IAEFluidStack> list = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+	private final ArrayList<IAEFluidStack> view = new ArrayList<>();
+	private final IScrollSource src;
+	private final ISortSource sortSrc;
+
+	private int rowSize = 9;
+
+	private String searchString = "";
+	private IPartitionList<IAEFluidStack> myPartitionList;
+	private String innerSearch = "";
+	private boolean hasPower;
+
+	public FluidRepo( final IScrollSource src, final ISortSource sortSrc )
+	{
+		this.src = src;
+		this.sortSrc = sortSrc;
+	}
+
+	public void updateView()
+	{
+		this.view.clear();
+
+		this.view.ensureCapacity( this.list.size() );
+
+		final Enum viewMode = this.sortSrc.getSortDisplay();
+		final Enum searchMode = AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_MODE );
+		final boolean needsZeroCopy = viewMode == ViewItems.CRAFTABLE;
+
+		this.innerSearch = this.searchString;
+		final boolean terminalSearchToolTips = AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_TOOLTIPS ) != YesNo.NO;
+
+		boolean searchMod = false;
+		if( this.innerSearch.startsWith( "@" ) )
+		{
+			searchMod = true;
+			this.innerSearch = this.innerSearch.substring( 1 );
+		}
+
+		Pattern m;
+		try
+		{
+			m = Pattern.compile( this.innerSearch.toLowerCase(), Pattern.CASE_INSENSITIVE );
+		}
+		catch( final Throwable ignore )
+		{
+			try
+			{
+				m = Pattern.compile( Pattern.quote( this.innerSearch.toLowerCase() ), Pattern.CASE_INSENSITIVE );
+			}
+			catch( final Throwable __ )
+			{
+				return;
+			}
+		}
+
+		boolean notDone = false;
+		for( IAEFluidStack fs : this.list )
+		{
+			if( this.myPartitionList != null )
+			{
+				if( !this.myPartitionList.isListed( fs ) )
+				{
+					continue;
+				}
+			}
+
+			if( viewMode == ViewItems.CRAFTABLE && !fs.isCraftable() )
+			{
+				continue;
+			}
+
+			if( viewMode == ViewItems.STORED && fs.getStackSize() == 0 )
+			{
+				continue;
+			}
+
+			final String dspName = searchMod ? Platform.getModId( fs ) : Platform.getFluidDisplayName( fs );
+			boolean foundMatchingFluidStack = false;
+			notDone = true;
+
+			if( m.matcher( dspName.toLowerCase() ).find() )
+			{
+				notDone = false;
+				foundMatchingFluidStack = true;
+			}
+
+			if( terminalSearchToolTips && notDone && !searchMod )
+			{
+				final List<String> tooltip = Platform.getTooltip( fs );
+
+				for( final String line : tooltip )
+				{
+					if( m.matcher( line ).find() )
+					{
+						foundMatchingFluidStack = true;
+						notDone = false;
+						break;
+					}
+				}
+			}
+
+			if( foundMatchingFluidStack )
+			{
+				if( needsZeroCopy )
+				{
+					fs = fs.copy();
+					fs.setStackSize( 0 );
+				}
+
+				this.view.add( fs );
+			}
+		}
+
+		final Enum SortBy = this.sortSrc.getSortBy();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)